processing/src/main/java/org/apache/druid/error/DruidException.java

```diff
@@ -0,0 +1,389 @@
+
+  public Category getCategory()
+  {
+    return category;
+  }
+
+  public String getErrorCode()
+  {
+    return errorCode;
+  }
+
+  public String getContextValue(String key)
+  {
+    return context.get(key);
+  }
+
+  public Map<String, String> getContext()
+  {
+    return context;
+  }
+
+  public int getStatusCode()
+  {
+    return category.getExpectedStatus();
+  }
+
+  /**
+   * Returns this DruidException as an ErrorResponse.  This method exists for compatibility with some older code
+   * paths that serialize out Exceptions directly using Jackson.  Instead of serializing a DruidException
+   * directly, code should be structured to take the DruidException and build an ErrorResponse from it to be
+   * used to push across the wire.
+   * <p>
+   * As such, this method should be deleted in some future world.  Anyone wondering how to serialize and deserialize
+   * a DruidException should look at {@link ErrorResponse} and leverage that instead of this.
+   *
+   * @return an ErrorResponse
+   */
+  @JsonValue
+  public ErrorResponse toErrorResponse()
+  {
+    return new ErrorResponse(this);
+  }
+
+  /**
+   * Builds a new DruidException with a message that is the result of prepending the message passed as a parameter
+   * with the message already on the DruidException.
+   *
+   * @param msg  Message to be prepended, can be a Java format string
+   * @param args Arguments to be passed to the message if it is a Java format string
+   * @return a new DruidException with prepended-message
+   */
+  public DruidException prependAndBuild(String msg, Object... args)
+  {
+    return new DruidException(
+        this,
+        errorCode,
+        targetPersona,
+        category,
+        StringUtils.format("%s: %s", StringUtils.nonStrictFormat(msg, args), getMessage())
+    ).withContext(context);
+  }
+
+  /**
+   * The persona that the message on a DruidException is targetting
+   */
+  public enum Persona
+  {
+    /**
+     * Represents the end-user, a persona who is issuing queries to the Druid Query APIs
+     */
+    USER,
+    /**
+     * Represents an administrative user, a persona who is interacting with admin APIs and understands Druid query
+     * concepts without necessarily owning the infrastructure and operations of the cluster
+     */
+    ADMIN,
+    /**
+     * Represents a persona who actively owns and operates the cluster.  This persona is not assumed to understand
+     * Druid query concepts, but instead understand cluster operational concepts.
+     */
+    OPERATOR,
+    /**
+     * Represents someone who has all of the context and knowledge to be actively diving into the Druid codebase.
+     * This persona exists as a catch-all for anything that is so deep and technically in the weeds that it is not
+     * possible to make a message that will make sense to a different persona.  Generally speaking, there is a hope
+     * that only DEFENSIVE error messages will target this persona.
+     */
+    DEVELOPER
```

Comment:
It might be "file a bug" or it might just be "this error message is completely impenetrable, we know that, but don't have the time to make it so that we can make a useful error message, so we just mark it as intended for the DEVELOPER audience and move on".  It would be great if there are no errors for this persona (or if they are basically all just DEFENSIVE exceptions)...